tests/Doctrine/Migrations/Tests/Stub/CustomClassNameMigrationFactory.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Tests\Stub;
+
+use Doctrine\Migrations\AbstractMigration;
+use Doctrine\Migrations\Version\MigrationFactory;
+
+class CustomClassNameMigrationFactory implements MigrationFactory
+{
+    /** @var MigrationFactory */
+    private $parentMigrationFactory;
+
+    /** @psalm-var class-string<CanItBeAnyClassOrShouldItBeASpecificType> */
```

Comment:
You will need to adapt my suggestion :laughing:

Reply:
:exploding_head:

Reply:
done